sway-fmt-v2/src/items/item_abi.rs

```diff
@@ -1,8 +1,97 @@
-use crate::fmt::{Format, FormattedCode, Formatter};
-use sway_parse::ItemAbi;
+use crate::{
+    config::items::ItemBraceStyle,
+    fmt::{Format, FormattedCode, Formatter},
+    utils::{attribute::FormatDecl, bracket::CurlyBrace},
+};
+use sway_parse::{token::Delimiter, AttributeDecl, ItemAbi};
+use sway_types::Spanned;
 
 impl Format for ItemAbi {
-    fn format(&self, _formatter: &mut Formatter) -> FormattedCode {
-        todo!()
+    fn format(&self, formatter: &mut Formatter) -> FormattedCode {
+        let mut formatted_code = String::new();
+
+        // Add enum token
+        formatted_code.push_str(self.abi_token.span().as_str());
+        formatted_code.push(' ');
+
+        // Add name of the abi
+        formatted_code.push_str(self.name.as_str());
+        Self::open_curly_brace(&mut formatted_code, formatter);
+
+        // Add item fields
+        // abi_items
+        formatted_code += self
+            .abi_items
+            .clone()
+            .into_inner()
+            .iter()
+            .map(|item| -> FormattedCode {
+                let mut buf = String::new();
```

Comment:
For indentation to be added, you can add something like

`buf.push_str(&formatter.shape.indent.to_string(formatter)); ` 

so that the current level of indentation is pushed before the formatted code pieces.

I am following a similar approach at [structs](https://github.com/FuelLabs/sway/blob/60da558a85563ddb5ffb4cffd46563c13b0d5935/sway-fmt-v2/src/items/item_struct.rs#L128) and [enums](https://github.com/FuelLabs/sway/blob/bdfb7449cac1e80ee4258abf1882c3caec99111c/sway-fmt-v2/src/items/item_enum.rs#L51) as well.